internal/client/connutils.go

```diff
@@ -1217,3 +1363,128 @@ func findEdgeDBTOML(paths *cfgPaths) (string, error) {
 		return tomlPath, nil
 	}
 }
+
+func jwtBase64Decode(data []byte) (map[string]interface{}, error) {
+	decoded := make([]byte, base64Encoding.DecodedLen(len(data)))
+	_, err := base64Encoding.Decode(decoded, data)
+	if err != nil {
+		return nil, err
+	}
+
+	var jwt map[string]interface{}
+	err = json.Unmarshal(decoded, &jwt)
+	if err != nil {
+		return nil, err
+	}
+
+	return jwt, nil
+}
+
+func (r *configResolver) parseCloudInstanceNameIntoConfig(
+	source string,
+	paths *cfgPaths,
+) (e error) {
+	if r.instance.val == nil {
+		return fmt.Errorf("missing instance")
+	}
+
+	if r.org.val == nil {
+		return fmt.Errorf("missing org")
+	}
+
+	var secretKey string
+	if r.secretKey.val != nil {
+		secretKey = r.secretKey.val.(string)
+	} else {
+		errMsg := "Cannot connect to cloud instances without secret key: %w"
+
+		dir, err := paths.CfgDir()
+		if err != nil {
+			return fmt.Errorf(errMsg, err)
+		}
+
+		profile := "default"
+		if r.profile.val == nil {
+			p, ok := os.LookupEnv("EDGEDB_CLOUD_PROFILE")
+			if ok {
+				r.setProfile(p, "EDGEDB_CLOUD_PROFILE environment variable")
+			}
```

Comment:
Sorry this was my bad in a previous version of the edgedb-python PR - `EDGEDB_CLOUD_PROFILE` should be set just as other connection params do (looks like you've already done that), not here.

Reply:
Hm... If this is not supposed to be here maybe [this test case](https://github.com/edgedb/shared-client-testcases/blob/main/connection_testcases.json#L2784) needs to be revised? Without this `LookupEnv()` call environment variables are only read if there is no host or port or instance or dsn or credentials file name or credentials json blob. However the test case sited seems to expect the environment to be read even though there is an instance set. Maybe I'm misunderstanding something.

Reply:
You're totally right, I did something like [this](https://github.com/edgedb/edgedb-python/pull/405/commits/c4383ecc07cb03805ce94f1cc28f555aca365dbd#diff-74188b7ca9fbcd019d3646664133b00e705e3453eb08dfc1726dde21a1e29f3bR527-R585) in edgedb-python to pass this test - get env var of the profile in the options layer, so that it's both functional on the options layer and the env var layer, and thus overwrites deeper layers like project stashed `cloud-profile`. I don't know maybe we should just allow setting cloud profile in the API options too (and fix this test to expect a failure) 🤷‍♂️ so that we have no exceptions like this.

Reply:
Seems like this type of special case can make the API confusing to use.

Reply:
Right, let's fix it then?

Reply:
Sure, I'll change this test to be a failure for now. Then we can add the secretKey option later.

Reply:
https://github.com/edgedb/shared-client-testcases/pull/22